### PR TITLE
Add documentation for 3001 errors

### PR DESCRIPTION
**Pull request by Morriar**
image

### CODE REVIEW

File: website/docs/error-reference.md [hunk 1]
```diff
@@ -53,6 +53,47 @@ and should not be reported.
...
+Sometimes, `EXPRESSION` is not a constant literal like in what follows:
+
+```rb
+class << some_variable
+  include Foo
+end
+```
+
+In this case, it is possible to directly call `include` on the the singleton
+class of `EXPRESSION`, but it should be done with **utmost caution**, as Sorbet
```
Comment (Morriar) [inline website/docs/error-reference.md:89]: The `utmost caution` part is copied from https://sorbet.org/docs/error-reference#4002

Comment (froydnj) [reply website/docs/error-reference.md:89]: Thanks for flagging this.

Nitpicking, I think we should revisit the language here and in the 4002 documentation, "far less accurate predictions" sounds vaguely like Sorbet is consulting a crystal ball -- maybe something like "make Sorbet's analyses far less accurate" would be better?  Regardless, this is followup material.

**Review by froydnj — APPROVED** [review]
Thank you!

**Comment by jez** [comment]
Thanks!

**Pull request merged**